crates/engine/tree/src/tree/mod.rs

```diff
@@ -1319,6 +1319,9 @@ where
             match self.insert_block(child) {
                 Ok(res) => {
                     debug!(target: "engine", child =?child_num_hash, ?res, "connected buffered block");
+                    if self.is_sync_target_head(child_num_hash.hash) {
+                        self.make_canonical(child_num_hash.hash);
+                    }
```

Comment:
this makes perfect sense, but I think we should add an additional check here that ensures, this is a new block by checking the InsertPayloadOk

https://github.com/paradigmxyz/reth/blob/017372516e1cb7685becd67c5be47f176664d2e5/crates/blockchain-tree-api/src/lib.rs#L256-L257

separately, we should consider not reusing this from legacy tree but roll our own that no longer needs the `BlockAttachment`

Reply:
> this makes perfect sense, but I think we should add an additional check here that ensures, this is a new block by checking the InsertPayloadOk

makes sense, done here https://github.com/paradigmxyz/reth/pull/10131/commits/b5b1b34770aa6d92821f61fb9cec253469a49994

> separately, we should consider not reusing this from legacy tree but roll our own that no longer needs the `BlockAttachment`

sure I can work on this in a follow-up